sklearn/utils/_array_api.py

```diff
@@ -822,6 +822,84 @@ def _estimator_with_converted_arrays(estimator, converter):
     return new_estimator
 
 
+def make_converter(X):
+    """Helper to implement the 'y follows X' rule.
+
+    Returns a function that converts an array to the namespace and device of X.
+
+    When X is not an array api array, the converter does nothing.
+    """
+    if isinstance(X, numpy.ndarray):
+
+        def convert(data):
+            if not hasattr(data, "__dlpack__"):
+                return data
+            data_xp, _ = get_namespace(data)
+            return _convert_to_numpy(data, data_xp)
+
+        return convert
+    xp, is_array_api, device_ = get_namespace_and_device(X)
+    if not is_array_api:
+
+        def convert(data):
+            return data
+
+        return convert
+
+    def convert(data):
+        if not isinstance(data, numpy.ndarray) and not hasattr(data, "__dlpack__"):
+            return data
+        data_xp, _, data_device = get_namespace_and_device(data)
+        if data_xp == xp and data_device == device_:
+            return data
+        try:
+            return xp.asarray(data, device=device_)
+        except Exception:  # pragma: no cover
+            # direct conversion to a different library may fail in which
+            # case we try converting to numpy first
+            data = _convert_to_numpy(data, data_xp)
+            return xp.asarray(data, device=device_)
+
+    return convert
+
+
+def convert_attributes(estimator, ref_array):
```

Comment:
```suggestion
def convert_attributes(estimator, reference_array):
```

or maybe just `reference`? I somehow prefer to spell out things instead of using abbrvs (abbreviations) because it is easier to read.